### PR TITLE
Improve performance in CSVOutputFormat

### PR DESCRIPTION
**Pull request by anthonyccri**
* Use Guava Escapers to escape embedded quotes in attributes rather
  than String.replace as the latter recompiles the regex for every
  single attribute
* Compute the attribute formatters once and reuse rather than checking
  instanceof for every attribute

### CODE REVIEW

File: src/wfs/src/test/java/org/geoserver/wfs/response/CSVOutputFormatTest.java [hunk 1]
```diff
@@ -5,17 +5,13 @@
...
 
-import static org.junit.Assert.*;
-
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.StringReader;
-import java.util.ArrayList;
-import java.util.List;
-
+import au.com.bytecode.opencsv.CSVReader;
```
Comment (aaime) [inline src/wfs/src/test/java/org/geoserver/wfs/response/CSVOutputFormatTest.java]: Copyright header should say "2014 - 2016"

File: src/wfs/src/main/java/org/geoserver/wfs/response/CSVOutputFormat.java [hunk 2]
```diff
@@ -5,17 +5,8 @@
  */
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/response/CSVOutputFormat.java]: Copyright header should say "2014 - 2016" now (from 2014 - 2015)

**Comment by aaime** [comment]
Since this is a performance improvement, can you share how much the performance did improve in your tests? :-)

**Comment by anthonyccri** [comment]
@aaime I fixed the copyright issue.  As far as quantifying the performance gain, I ran a test of serializing 500K features, each with 50 attributes, using the old code and the new code and observed a performance gain of 2.5X.  On my laptop, this dropped processing time from 25s to 10s.

**Comment by aaime** [comment]
Please next time open a ticket and refer it from the commit message. I've created the ticket for you, so that this commit makes into the release notes: https://osgeo-org.atlassian.net/browse/GEOS-7406

**Pull request merged**